### PR TITLE
Julia 0.6 fixes

### PR DESCRIPTION
**Pull request by andreasnoack**
Is there a better way to handle the new `where` syntax than `eval(parse(...))`?

### CODE REVIEW

File: src/balanced_tree.jl [hunk 1]
```diff
@@ -119,21 +119,21 @@ type BalancedTree23{K, D, Ord <: Ordering}
...
     freedatainds::Array{Int,1}
-    useddatacells::IntSet 
+    useddatacells::IntSet
     # The next two arrays are used as a workspace by the delete!
     # function.
     deletionchild::Array{Int,1}
     deletionleftkey::Array{K,1}
     function BalancedTree23(ord1::Ord)
-        tree1 = Array(TreeNode{K}, 1)
+        tree1 = Array{TreeNode{K}}(1)
```
Comment (tkelman) [inline src/balanced_tree.jl]: should use Vector when the dimension is know to be 1

Comment (andreasnoack) [inline src/balanced_tree.jl]: I don't really see why that is important. It is implied by the number of arguments.

Comment (tkelman) [inline src/balanced_tree.jl]: the compiler has to do more work if it's not explicit

**Comment by andreasnoack** [comment]
Updated and rebased. I've also enabled testing on nightly to avoid that we break the package again.

**Pull request merged**